scripts/zones/Maze_of_Shakhrami/npcs/Rockwell.lua

```diff
@@ -9,23 +9,31 @@ local ID = zones[xi.zone.MAZE_OF_SHAKHRAMI]
 local entity = {}
 
 entity.onTrade = function(player, npc, trade)
-    if
-        player:getQuestStatus(xi.quest.log_id.JEUNO, xi.quest.id.jeuno.YOUR_CRYSTAL_BALL) == QUEST_ACCEPTED and
-        npcUtil.tradeHas(trade, xi.item.AHRIMAN_LENS)
-    then
-        player:setCharVar('QuestYourCrystalBall_prog', 1)
-        player:confirmTrade(trade)
+    if player:getQuestStatus(xi.quest.log_id.JEUNO, xi.quest.id.jeuno.YOUR_CRYSTAL_BALL) == QUEST_ACCEPTED then
+        if npcUtil.tradeHasExactly(trade, { { xi.item.AHRIMAN_LENS, 1 } }) then
+            if player:getCharVar('QuestYourCrystalBall_prog') > os.time() then
+                player:messageSpecial(ID.text.CANNOT_SUBMERGE, xi.item.AHRIMAN_LENS)
+            elseif player:getCharVar('QuestYourCrystalBall_prog') == 0 then -- don't initiate submerge if previous one hasn't been collected
+                player:confirmTrade()
+                player:setCharVar('QuestYourCrystalBall_prog', os.time() + 30) -- current retail capture shows 30s time to pickup
+                player:messageSpecial(ID.text.SUBMERGE, xi.item.AHRIMAN_LENS)
+            end
+        elseif npcUtil.tradeHas(trade, xi.item.AHRIMAN_LENS) then
+            player:messageSpecial(ID.text.CANNOT_SUBERMGE, xi.item.AHRIMAN_LENS)
+        end
     end
 end
 
 entity.onTrigger = function(player, npc)
     if
         player:getQuestStatus(xi.quest.log_id.JEUNO, xi.quest.id.jeuno.YOUR_CRYSTAL_BALL) == QUEST_ACCEPTED and
-        player:getCharVar('QuestYourCrystalBall_prog') == 1
+        player:getCharVar('QuestYourCrystalBall_prog') ~= 0
     then
-        player:startEvent(52)
-    else
-        player:messageSpecial(ID.text.NOTHING_OUT_OF_ORDINARY)
+        if player:getCharVar('QuestYourCrystalBall_prog') < os.time() then
```

Comment:
This is a bad condition block.  If you're checking less than, there's no need to explicitly check greater than which will also cause a corner case issue of not allowing exactly equal

Reply:
100%, not sure where this snuck in but you're right it isn't necessary given the ~=0 condition + < condition.. fixed!

Reply:
Thanks for the correction :)